### PR TITLE
implemented Order model with OrderDao and OrderService layers

### PR DESCRIPTION
**Pull request by MaliukDaria**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public List<Order> getUserOrders(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> getUserOrdersQuery = session.createQuery(
+                    "SELECT DISTINCT o FROM Order o "
+                            + "JOIN FETCH o.user "
+                            + "JOIN FETCH o.tickets t "
+                            + "JOIN FETCH t.movieSession ms "
+                            + "JOIN FETCH ms.movie "
+                            + "JOIN FETCH ms.cinemaHall "
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: Does it all work with inner join?

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                            + "WHERE o.user.id = :userId", Order.class)
+                    .setParameter("userId", user.getId());
+            return getUserOrdersQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException(
+                    "Can't get orders of user with id: " + user.getId(), e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: Is this catch block necessary?

File: src/main/java/com/dev/cinema/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    @JoinColumn(name = "order_id")
+    private List<Ticket> tickets;
+
+    @ManyToOne
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/model/Order.java:23]: Screenshot 2020-10-12 at 13 47 36

File: src/main/java/com/dev/cinema/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    private User user;
+
+    private LocalDateTime orderDate;
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/model/Order.java:26]: Do we need this empty line?

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order(tickets, user, LocalDateTime.now());
+        return add(order);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java]: What should happen to the shopping cart's tickets?

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: yes)

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: done

File: src/main/java/com/dev/cinema/model/Order.java (hunk 4 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/model/Order.java:26]: it exists only for readability

File: src/main/java/com/dev/cinema/model/Order.java (hunk 3 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/model/Order.java:23]: I used FetchType.LAZY in @OneToMany connection, with @ManyToOne we can use  FetchType.EAGER

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java (hunk 5 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java]: fixed

File: src/main/java/com/dev/cinema/model/Order.java (hunk 3 above)
Comment (dariiabutorina) [reply src/main/java/com/dev/cinema/model/Order.java:23]: My apologies, you're right

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**